count_cells.py

```diff
@@ -39,87 +43,87 @@ def __init__(self, argv):
             logging.getLogger().setLevel(logging.DEBUG)
 
-        project_dirs = get_target_project_dirs()
+        if self.args.accession:
+            accessions = [self.args.accession]
+        else:
+            accessions = [p.name for p in get_target_project_dirs()]
 
-        for project_dir in project_dirs:
-            self.count_one_project_cells(project_dir.name)
+        for accession in accessions:
+            cell_count = self.get_project_cell_count(accession)
+            if self.args.write:
+                self.write_cell_count(accession, cell_count)
 
-    def count_one_project_cells(self, accession_id: str):
+    @classmethod
+    def get_cached_cell_count(cls, accession: str) -> int:
         """
-        Count cells in one project
-
-        :param accession_id: An accession id with a downloaded matrix file
-        :param save_to_file: If true write the cell count total(s) to a global cell count file
+        Return a cell count for the given accession
         """
-        cell_count = self.get_project_cell_count(accession_id)
-        if self.args.write:
-            self.update_cell_count_file(accession_id, cell_count)
+        stats_file = Path(f'projects/{accession}/stats.json')
+        if stats_file.exists():
+            with open(stats_file, 'r') as f:
+                cell_count = json.loads(f.read()).get('cell_count', 0)
+            return cell_count
+        else:
+            return 0
 
     @classmethod
-    def get_accession_ids(cls) -> Sequence[str]:
+    def get_cached_cell_counts(cls) -> Mapping[str, int]:
         """
-        Return a list of the accession ids
+        Return a dict of accession id to cell count
         """
-        accession_ids = []
-        for p in Path('projects').iterdir():
-            logging.debug('Checking: %s', p)
-            if p.is_dir() and p.is_symlink():
-                accession_ids.append(p.name)
-                logging.debug('Found: %s', p.name)
-        return accession_ids
+        cell_counts = {}
+        for accession in [p.name for p in get_target_project_dirs()]:
+            cell_counts[accession] = cls.get_cached_cell_count(accession)
+        return cell_counts
 
     @classmethod
-    def get_cell_counts(cls):
-        if cls.cell_counts_file.exists():
-            with open(cls.cell_counts_file, 'r') as f:
-                cell_counts = json.loads(f.read())
-            return cell_counts
-        else:
-            return {}
-
-    def update_cell_count_file(self, accession_id: str, cell_count: int) -> bool:
+    def write_cell_count(cls, accession: str, cell_count: int) -> bool:
         """
-        Write the accession cell count to the global cell count file
+        Write the accession cell count to the project's stats json
 
-        :param accession_id: An accession id
-        :param cell_count: An int value to save, or None to remove entry
-        :return: Boolean status of the save
+        :param accession: An accession id
```

Comment:
Accessions **are** identifiers. And *ID* is an acronym (https://en.wikipedia.org/wiki/Identifier). So:

```suggestion
        :param accession: The accession of the project to write the cell count for.
```